### DISCUSSION TITLE
Nuxt 3.0 stable

### DISCUSSION THREAD

**Discussion by pi0** [nuxt/nuxt#15984] [Announcements]
We are thrilled to announce the first stable version of Nuxt 3.0.0 :sparkles:

Nuxt 3 is a modern rewrite of the Nuxt framework based on [Vite](https://vitejs.dev/), [Vue3](https://vuejs.org/), and [Nitro](https://nitro.unjs.io/) with first-class Typescript support and the result of more than two years of research, community feedback, innovation, and experiment to make a pleasant full-stack Developer Experience for Vue development to everyone.

- ✨ [Release Announcenment](https://nuxt.com/v3)
- 🐦 [Tweet](https://twitter.com/nuxt_js/status/1592904778337906689)
- 📝 [Changelog](https://github.com/nuxt/framework/releases/tag/v3.0.0)

**Comment by atodd** [comment]
Thank you Nuxt team!  Very excited to test out the first stable build!

**Comment by chrisbbreuer** [comment]
Congratulations! This is amazing! 🍾

**Comment by miracleonyenma** [comment]
**Awesome!!!!** 🎉🎉

**Comment by vanling** [comment]
Congratulations  on release and the new nuxt.com domain :D

**Comment by hausg** [comment]
Congratulations !!!

**Comment by vinayakkulkarni** [comment]
Can't wait to get my hands on it :)

**Comment by thomasbnt** [comment]
Congrats 🍾🚀🚀

**Comment by mrleblanc101** [comment]
@pi0 Any reason why `@nuxtjs/i18n` is not on the official roadmap ? 
But there is `nuxt/translations` on the roadmap which is only the documentation translation and not the translation module itself ?
`@nuxtjs/i18n` is an essential module for many people.
I know it's a community module, but it's still under the `@nuxtjs` namescpace and is as important as `@nuxtjs/auth` imho which is on the roadmap.

**Reply by pi0** [reply]
The roadmap in docs is focused on features we plan to partially add to either core or built-in dependencies.

@kazupon is hardly working on i18n support Check [intlify/vue-i18n-next](https://github.com/intlify/vue-i18n-next) and [intlify/nuxt3](https://github.com/intlify/nuxt3).

Discussion: https://github.com/nuxt/framework/discussions/9066

**Reply by sadeghbarati** [reply]
https://v8.i18n.nuxtjs.org/

**Comment by arnauddoub** [comment]
Congrats ! 🎉

**Comment by insru** [comment]
Congrats! Amazing project :)

**Comment by StanleyMasinde** [comment]
Great job!

**Comment by Kyriakos-Michael** [comment]
As an old man said, the Nuxt big thing!

**Comment by CodyBontecou** [comment]
Huge release today. Great job to the Nuxt team!

**Comment by AllanOricil** [comment]
Amazing work. Thank you

**Comment by FileSoft-us** [comment]
Let's go baby

**Comment by warmthsea** [comment]
Thanks

**Comment by Julien-R44** [comment]
Well done guys! For Nuxt, and all things you've built next to it. Volta, Docus, all the cool modules from unjs... What you are doing for the ecosystem is awesome. Thanks a lot for that! 🙌

**Comment by RajendraUppu** [comment]
Started using it from first release candidate since then I've seen Nuxt team Efforts on 14 Rc Releases but Finally Made it✌️ **Nuxt 3 Stable Release** 
**Kudos** To you **#NuxtTeam** ❤️🔥🙏🏻

**Comment by zorenkonte** [comment]
Nuxt team rocks!🤘

**Comment by rahulkumarsingh73690** [comment]
```
<script>
export default {
  head() {
    return {
          title: ``,
           }
         }
}
<script>
```

Head tag doesn't return data after updating to nuxt3 stable or rc-14. Only useHead() works with setup why?

**Reply by pi0** [reply]
You need `defineNuxtComponent` since RC.14. Checkout https://github.com/nuxt/framework/pull/8901

**Reply by rahulkumarsingh73690** [reply]
@pi0 Sorry but I can't find proper information in this pr or docs. I am using not using script setup on some pages. Can you please tell me how to use defineNuxtComponent instead of vue 2 head()?

**Comment by duncandejong** [comment]
Thanks for all the hard work!

**Comment by itzaks** [comment]
Amazing work! ✨

**Comment by scottyzen** [comment]
Congratulations, Amazing work! 🚀

**Comment by chantouchsek** [comment]
Congrats, Thanks nuxt team for the big release today.

**Comment by kstraszewski** [comment]
Great work!
But i do have this error after upgrading from rc9:
`node_modules/nuxt/dist/app/config.mjs:3:24: ERROR: Could not resolve "#build/app.config.mjs"`
I see that this is error with #imports or #app but I am lost which version of import should I use?

**Comment by ThitipongInlom** [comment]
Amazing work! ✨

**Comment by lucassimines** [comment]
Thanks you!!! Nuxt is AWESOME 💚

**Comment by rahulkumarsingh73690** [comment]
Is @nuxtjs/PWA planned for nuxt 3?

**Reply by henrycunh** [reply]
this has landed already
https://github.com/kevinmarrec/nuxt-pwa-module

**Reply by rahulkumarsingh73690** [reply]
This is not working properly.

**Reply by rahulkumarsingh73690** [reply]
This is not working properly.

**Reply by henrycunh** [reply]
what problem are you having?

**Comment by vell174** [comment]
Hi Thanks you!!! Will nuxt 3 work in ie11 ?

**Reply by kissu** [reply]
[IE is dead](https://death-to-ie11.com/) so don't expect Vue or Nuxt to support it in the future (at all).

**Comment by moabdurrakib** [comment]
Congratulations. Amazing works.

**Comment by therealokoro** [comment]
Finally. For a while there I thought "summer 2022" was actually a typo in the announcements....Maybe y'all meant to say summer 2023 😅😅😅

**Comment by akasection** [comment]
is it ok to replace `latest` nuxt npm tag to 3.0.0? 

I mean, there's still no nuxt 2 tag on the npm

**Comment by sarath-aj** [comment]
I waited for this so long

**Comment by icarusgk** [comment]
Been waiting for this release 🥳 Thanks a bunch Nuxt Team. Using this as much as I can for my personal project from now on lol

**Comment by maxdzin** [comment]
That's amazing!
Big Love for the Nuxt team and community 💚

**Comment by danwritecode** [comment]
Great work guys! been using Nuxt 3 since the very beginning. I actually just upgraded an old project all the way from RC-8 to 3.0.0 today...it felt good!

**Comment by ash1khan** [comment]
🎉 Finally. That's an insane amount of effort and monumental task guys. Hat's off for the things you guys brought to solo freelancer like me and compel me to stay in loved with nuxt... 🎊

**Comment by Amirali-Yavari** [comment]
i love u ❤️ Nuxt 
@pi0 
@antfu

**Comment by Envoy49** [comment]
Does this mean that you are not going to support NUXT Bridge?

**Comment by Sandlens** [comment]
Huge congratulations to the Nuxt team!!

**Comment by Zerro97** [comment]
🎉🎉🎉🎉 Congrats! 🎉🎉🎉🎉

**Comment by jamesray** [comment]
🍾 Congrats Nuxt @pi0 @danielroe @antfu @yyx990803 @Atinux et al 🙇🏻 🙇🏻 🙇🏻 !

**Comment by andysay** [comment]
thx for work!

**Comment by zhaoweizhong** [comment]
Nice work! Then how about the Auth module?

**Comment by hrgweb** [comment]
Love it! Thanks guys!

**Comment by uyamazak** [comment]
Excellent work so far! thanks!
I have already been using it for over 6 months in my job and it is a great framework!

**Comment by hieu-ht** [comment]
Awesome, congratulations <3

**Comment by liulib** [comment]
Congrats! 🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉

**Comment by feelixe** [comment]
Congratulations and thanks for your hard work! 🎉

**Comment by zaeval** [comment]
Congratulations and thanks for your hard work! 🎉!

**Comment by catalin560** [comment]
about time I'd say... I was constantly watching the roadmap which said the stable release would come out in spring and then in summer 😆

**Comment by maxtomczyk** [comment]
Finally, after 7 months of developing product with release candidate versions I can install stable version <3
Congratulations!

**Comment by manzoliric** [comment]
awesome, thanks for your work!  😃

**Comment by Redemption198** [comment]
Congrats!

**Comment by eveningforgery** [comment]
Congrats and thanks to all contributors! 🎉🎉🎉

**Comment by cesswhite** [comment]
Congrats and thank you for all! Amazing work 🎉

**Comment by targetlucked69** [comment]
Awesome! thank you ffor all your efforts

... where did the `watch` property from nuxt config go? How can we watch files for changes now?

**Comment by adrianrivers** [comment]
Fantastic news, congratulations 😊

**Comment by fskarmeta** [comment]
Congrats :) !

**Comment by alterhu2020** [comment]
Congrats! amazing work done! 🎉🎉🎉

**Comment by JenuelDev** [comment]
Nuxt and Vue 3 are the Best!!! <3 <3 <3

**Comment by ben890524** [comment]
Congratulations 🎉

**Comment by kotering** [comment]
Congratulations!
I'm looking forward to implementing new applications with Nuxt3.

**Comment by MuhammadKhizar7** [comment]
Congratulations!

**Comment by mittci** [comment]


**Reply by Redemption198** [reply]
Don't use this discussion to post issues, it probably won't be noticed. Open an issue instead, before that try to run `npx nuxi upgrade -f` and see if it solves your issue.

**Reply by mittci** [reply]
all is ok, simply useBody has been renamed to readBody

**Reply by Redemption198** [reply]
oh ok

**Comment by yeonjulee1005** [comment]
Congratulations! 💯

**Comment by modelair** [comment]
Congrats!

**Comment by favourwright** [comment]
I only noticed when I visited v3.nuxtjs.org documentation and was redirected to nuxt.com two days ago...
this release definitely warms my heat♥

I saw a PR relating to `default scroll behavior`
question: is the scroll behavior now fixed and we don't need to use `/app/router.options.js`

**Comment by zavvla** [comment]
@pi0  Congratulations!!! 
 
Everyone was really looking forward to this day =)
Tell me, will there be documentation for other languages, as it was with the second version?

**Comment by 1x1-lab** [comment]
up up  🎉  🎉 !!!!

**Comment by pagamini** [comment]
i'd like it😄

**Comment by wikanonymous-dev** [comment]
Congrats!!! 🎉

**Comment by Disorrder** [comment]
hype hype hype!!! 🎉
Thanks for perfect framework!

**Comment by nazarepiedady** [comment]
When will be it possible to start translating the documentation to other languages?

**Reply by nazarepiedady** [reply]
@danielroe do you have an answer to my question?

**Reply by danielroe** [reply]
Currently @pi0 is planning/working on this. See https://nuxt.com/docs/community/roadmap:

> [nuxt/translations#4](https://github.com/nuxt/translations/discussions/4) ([request access](https://github.com/nuxt/framework/discussions/765))
>
> A collaborative project for a stable translation process for Nuxt 3 docs. Currently pending for ideas and documentation tooling support (content v2 with remote sources).

**Comment by vedmant** [comment]
When we can expect documentation for Options API usage with Nuxt 3?

**Reply by danielroe** [reply]
Here you go: https://nuxt.com/docs/api/utils/define-nuxt-component#definenuxtcomponent

**Reply by vedmant** [reply]
@danielroe There is no examples of how to use things like route params, runtime configs, redirects, errors, API usage (axios/fetch), Vuex migration, Pinia usage, cookies, layouts, basically everything that was moved to use* compostables.

**Reply by InFecTie** [reply]
@danielroe  is there already better documentation for options API migration from nuxt2 to nuxt3 like asked above?
'....API usage (axios/fetch) and basically everything that was moved to use* compostables.'

Doing a migration with a big solution from nuxt2 is still a pain in the ass and very demotivating to actually transition to nuxt3

**Reply by vedmant** [reply]
Yeah, when Vue introduced composable API some devs decided that options API is deprecated and switched to composition, but that's not the case, options API was always superior in terms of maintainability and code quality. I move my projects to Nuxt 3, but keep all in options API, however lack of documentation for options API in Nuxt makes this Nuxt incomplete and kind if forces people to use composition API, which is a big mistake.

**Comment by noopurphalak** [comment]
Amazing work guys...the hard work as finally paid off for you guys...!!! Can't wait to play around with nuxt 3...
